examples/aya-tool/xtask/src/build_ebpf.rs

```diff
@@ -52,8 +51,14 @@ pub fn build_ebpf(opts: Options) -> Result<(), anyhow::Error> {
         "--profile",
         opts.profile.as_str(),
     ];
+
+    // Command::new creates a child process which inherits all env variables. This means env
+    // vars set by the cargo xtask command are also inherited. RUSTUP_TOOLCHAIN is removed
+    // so the rust-toolchain.toml file in the -ebpf folder is honored.
+
     let status = Command::new("cargo")
         .current_dir(&dir)
+        .env_remove("RUSTUP_TOOLCHAIN")
```

Comment:
I think you should do this in a separate PR, so when nightly is fixed we can revert this one and still keep this fix

Reply:
What about separate commit?

Reply:
I've pushed it as a separate commit. I don't really want to submit a new PR, because a standalone "pin nightly" PR would need to modify the same file to replace `+nightly` with `+nightly-2023-01-10`, quite a hassle IMO.

A separate commit is going to be easy enough to revert.